src/index.js

```diff
@@ -1,28 +1,48 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import {BrowserRouter} from "react-router-dom";
-import {Provider} from "react-redux";
+import React from "react";
+import ReactDOM from "react-dom";
+import { BrowserRouter } from "react-router-dom";
+import { Provider } from "react-redux";
 import axios from "axios";
 
-import PokemonApp from './PokemonApp';
 import store from "./redux/store";
-import {fetchPokemonTypes} from "./redux/pokemonTypesSlice";
+import { PokemonApp } from "./PokemonApp";
 
-axios.interceptors.response.use(function (response) {
+axios.interceptors.response.use(
+  function (response) {
+    if (response.status === 201) {
+      window.location.href = "/";
+    }
     return response.data;
-}, function (error) {
+  },
+  function (error) {
+    if (error.response.status === 401) {
+      localStorage.removeItem("token");
+      window.location.href = "/login";
```

Comment:
Please reconsider this in the context of private routes, probably it can be changed

Reply:
There's an issue with the unauthorized user. When they try to get access to a restricted unauthorized users page (f.e. '/pokemon') API sends a response with status 401. It is implemented with JWT strategy API. The point is a valid token. Even if the user changes the token in local storage manually, JWT strategy will not allow getting access.